### PR TITLE
namespace command

### PR DESCRIPTION
**Pull request by cagiti**
- feat: namespaces command
- fix: refactor and added tests

### CODE REVIEW

File: pkg/cmd/namespacecmd/namespace.go [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+func (c *NamespaceCmd) Run() error {
+	d := domain.NewNamespace()
+
+	err := d.Validate()
+	if err != nil {
+		return errors.Wrap(err, "validate failed")
+	}
+	err = d.Run()
+	if err != nil {
+		return errors.Wrap(err, "validate failed")
```
Comment (macox) [inline pkg/cmd/namespacecmd/namespace.go:45]: "run failed"

File: pkg/kube/factory.go [hunk 2]
```diff
@@ -27,19 +29,78 @@ func (f *factory) SetKubeContext(context string, config *api.Config) (*api.Confi
...
+		name := config.CurrentContext
+		if name != "" && config.Contexts != nil {
+			return config.Contexts[name]
+		}
+	}
+	return nil
+}
 
-	return config, nil
+func (f *factory) GetCurrentNamespace(config *api.Config) string {
```
Comment (macox) [inline pkg/kube/factory.go:97]: Is this used?

File: pkg/cmd/namespacecmd/namespace.go (hunk 1 above)
Comment (cagiti) [reply pkg/cmd/namespacecmd/namespace.go:45]: copy/pasta fun, cheers bro - will sort

File: pkg/kube/factory.go (hunk 2 above)
Comment (cagiti) [reply pkg/kube/factory.go:97]: possibly not, should be though - think I got lost in the moment - diolch

Comment (cagiti) [reply pkg/kube/factory.go:97]: yeah I should be using that, I forgot about that implementation, when selecting from the list it should display the current namespace at the top - great catch bud 👍

File: pkg/cmd/namespacecmd/namespace.go (hunk 1 above)
Comment (cagiti) [reply pkg/cmd/namespacecmd/namespace.go]: was the same in context command too

File: pkg/kube/factory.go (hunk 2 above)
Comment (cagiti) [reply pkg/kube/factory.go:97]: sorted

File: pkg/domain/context.go [hunk 3]
```diff
@@ -46,10 +46,10 @@ func (c *Context) Run() error {
 	return nil
 }
 
-func (c *Context) selectContext() (string, error) {
+func (c *Context) selectContext(currentContext string) (string, error) {
```
Comment (macox) [inline pkg/domain/context.go:49]: you may not need this param as you pass in c.Config.CurrentContext, and this func already has access to the c object

Comment (cagiti) [reply pkg/domain/context.go:49]: yeah was thinking that... but as this is only ever used it seemed silly to copy it

Comment (cagiti) [reply pkg/domain/context.go:49]: there is a better way though

Comment (cagiti) [reply pkg/domain/context.go]: nice one @macox, I've refactored based off your comment.

**Pull request merged**